test/unittesting/testingutils.go

```diff
@@ -0,0 +1,624 @@
+	return uint64(time.Now().UnixNano() / 1000000)
+}
+
+//helper function to execute shell commands
+func shellout(command string) {
+	installationPath := getInstallationDir()
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.Dir = installationPath
+	err := cmd.Start()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func stopST(pid string) {
+	installationPath := getInstallationDir()
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	pidsBefore := getListOfPids()
+	if len(pidsBefore) == 0 {
+		return
+	}
+	if len(pidsBefore) == 1 {
+		if pidsBefore[0] == "" {
+			return
+		}
+	}
+	cmd := exec.Command("kill", pid)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.Dir = installationPath
+	err := cmd.Run()
+
+	if err != nil {
+		log.Fatal(err.Error(), "error in killSt")
+	}
+
+	startTime := getCurrTimeInMS()
+	for getCurrTimeInMS()-startTime < 10000 {
+		pidsAfter := getListOfPids()
+		if itemExists(pidsAfter, pid) {
+			time.Sleep(100 * time.Millisecond)
+		} else {
+			return
+		}
+	}
+	panic("Could not stop ST")
+}
+
+func itemExists(arr []string, item string) bool {
+	for i := 0; i < len(arr); i++ {
+		if arr[i] == item {
+			return true
+		}
+	}
+	return false
+}
+
+func CleanST() {
+	installationPath := getInstallationDir()
+
+	cmd := exec.Command("rm", "config.yaml")
+	cmd.Dir = installationPath
+
+	err := cmd.Run()
+
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the config yaml file [THIS IS CLEAN-ST]")
+	}
+
+	cmd = exec.Command("rm", "-rf", ".webserver-temp-*")
+	cmd.Dir = installationPath
+	err = cmd.Run()
+
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the webserver-temp files [THIS IS CLEAN-ST]")
+	}
+
+	cmd = exec.Command("rm", "-rf", ".started")
+	cmd.Dir = installationPath
+	err = cmd.Run()
+
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the .started file [THIS IS CLEAN-ST]")
+	}
+
+}
+
+func KillAllST() {
+	pids := getListOfPids()
+	for i := 0; i < len(pids); i++ {
+		stopST(pids[i])
+	}
+}
+
+/*this returns the number of slashes required to reach the to supertokens root assuming it is in the same level as supertokens-golang form the current working directory*/
+func returnNumberOfDirsToGoUpFromCurrentWorkingDir(isPathProvided bool) string {
+	mydir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err.Error(), "THIS IS RETURN-NUMBER-OF-DIRS-TO-GO-UP-FROM-CURRENT-WORKING-DIR")
+	}
+	arr := strings.Split(mydir, "/")
+	counter := 0
+	for i := 0; i < len(arr); i++ {
+		if arr[i] == "supertokens-golang" {
+			counter = i
+			break
+		}
+	}
+	var numberOfElems int
+	if isPathProvided {
+		numberOfElems = len(arr) - counter - 1
+	} else {
+		numberOfElems = len(arr) - counter
+	}
+	var dirUpSlash string
+	for i := 0; i < numberOfElems; i++ {
+		dirUpSlash += "../"
+	}
+	return dirUpSlash
+}
+
+func RemoveTrailingSlashFromTheEndOfString(input string) string {
```

Comment:
This function doesn't seem to be used anywhere. Remove it. Make sure to remove all unused functions?